src/Core/Scheduler/Providers/StaticProvider.js

```diff
@@ -0,0 +1,130 @@
+import { Vector3 } from 'three';
+import Extent from '../../Geographic/Extent';
+import OGCWebServiceHelper from './OGCWebServiceHelper';
+import Fetcher from './Fetcher';
+
+// select the smallest image entirely covering the tile
+function selectBestImageForExtent(images, extent) {
+    let selection;
+    for (const entry of images) {
+        if (extent.isInside(entry.extent)) {
+            if (!selection) {
+                selection = entry;
+            } else {
+                const d = selection.extent.dimensions();
+                const e = entry.extent.dimensions();
+                if (e.x <= d.x && e.y <= d.y) {
+                    selection = entry;
+                }
+            }
+        }
+    }
+    return selection;
+}
+
+
+/**
+ * This provider uses no protocol but instead download static images directly.
+ *
+ * It uses as input 'image_filename: extent' values and then tries to find the best image
+ * for a given tile using the extent property.
+ */
+export default {
+    preprocessDataLayer(layer) {
+        if (!layer.extent) {
+            throw new Error('layer.extent is required');
+        }
+
+        if (!(layer.extent instanceof Extent)) {
+            layer.extent = new Extent(layer.projection, ...layer.extent);
+        }
+
+        layer.options = {};
+        layer.canTileTextureBeImproved = this.canTileTextureBeImproved;
+        layer.url = new URL(layer.url, window.location);
+        return Fetcher.json(layer.url.href).then((metadata) => {
+            layer.images = [];
+            // eslint-disable-next-line guard-for-in
```

Comment:
eslint complains because you didn't check `if (metadata.hasOwnProperty(image))` inside the for-loop.

Note that you *might* need to use `Object.prototype.hasOwnProperty.call(metadata, image)` instead if it does not work, see https://eslint.org/docs/rules/guard-for-in

Reply:
The help page isn't helpful: `This rule is aimed at preventing unexpected behavior that could arise from using a for in loop` : Sure, but what "unexpected behavior" could possibly happen? (especially in this case where metadata is a json file?

Reply:
Yes, in this particular case, we don't really risk anything. It's just that I prefer to add a guard than to disable the linter. But I should really have prefixed this with `nits`, it's not very important.